### DISCUSSION TITLE
Query on Polymorphic stream

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#219] [Q&A]
I have similar usecase as below:
https://kb.timebase.info/community/development/qql/QQL%205.5/qql-tut-polymorphic

From documentation example:
As trades and orders will be 2 different datasets. So how to add both data in a polymorphic stream if the records are on same timestamp? (No example shown)

My understanding is that we should use WriteMode("INSERT") but I am unable to see this mode in tbapi (python) instead there is "REPLACE" mode 
 which is not mentioned in documentation.
as per:
https://kb.timebase.info/community/technology/apis/tickloader

Can you please suggest workaround?

Earlier in fixed stream for single dataset i manged to add same by having already sorted data and inserting by using "APPEND" mode.

But for polymorphic where I have 2 separate datasets I am having no clue how to add without INSERT Mode.
I am stuck with this, please help

**Comment by alex-karpovich** [comment]
INSERT Mode for python is disabled in latest versions. We will enable it in new version ASAP.

**Reply by ankitsoin** [reply]
Thanks for your reply.
In fixed stream also loading 300 mb file using spark streaming is causing issues as data is truncated as in between due to distributed platform of spark(only 7mil records are loaded instead of 15mil). 
I believe only Insert mode can only be the solution to my problem. Feel free to provide any suggestion if you may have.

My whole progress to timebase usage is halted.
Please help me knowing once updated in new version.

Thanks a lot for your efforts.

**Reply by alex-karpovich** [reply]
Actually possible workaround for now:
- you can load each dataset into separate stream
- TimeBase API allows to read several streams from single cursor, in this case messages will sort by time automatically

**Reply by ankitsoin** [reply]
Thanks for the workaround. Surely I will load separately for now.

But, to load huge data like 15-20 million to Timebase is still problematic for me in fixed stream. As data is getting truncated using append mode. I guess spark is shuffling the data due to which input data ain't remained sorted.
I guess this is specific use case of Insert mode.

Will wait for your next updated version.
Thanks in advance!!

**Reply by alex-karpovich** [reply]
https://github.com/epam/TimeBaseClientPython/releases/tag/6.0.3

**Reply by ankitsoin** [reply]
Superb!
Thanks a lot 👍
